citrination_client/models/service_status.py

```diff
@@ -0,0 +1,125 @@
+from citrination_client.models import Event
+from citrination_client.base.errors import CitrinationClientError
+
+class ServiceStatus(object):
+    """
+    The status of a service for a given model on Citrination. The status of a
+    service is summarized by the ready value indicating whether or not the service
+    is in a state where it can be utilized. It also includes more detailed
+    information about the current state of the service in the form of a descriptive
+    sentence, a context string and an event object.
+    """
+
+    @staticmethod
+    def from_response_dict(response_dict):
+
+        ServiceStatus._validate_response_dict(response_dict)
+
+        if "event" in response_dict:
+            event_dict = response_dict["event"]
+            event = Event(
+                title = event_dict["title"],
+                subtitle = event_dict["subtitle"],
+                normalized_progress = event_dict["normalizedProgress"]
+            )
+
+            if "subevent" in response_dict["event"]:
+                subevent_dict = response_dict["event"]["subevent"]
+                subevent = Event(
+                    title = subevent_dict["title"],
+                    subtitle = subevent_dict["subtitle"],
+                    normalized_progress = subevent_dict["normalizedProgress"]
+                )
+                event.subevent = subevent
+        else:
+            event = None
+
+        return ServiceStatus(
+            ready = response_dict["ready"],
+            reason = response_dict["reason"],
+            context = response_dict["context"],
+            event = event
+        )
+
+    @staticmethod
+    def _validate_response_dict(response_dict):
```

Comment:
Why only apply this when building `from_response_dict`?  A future developer is free to create one of these objects with `__init__`, and might be confused when they don't get member validation.